includes/object-cache.php

```diff
@@ -679,7 +683,13 @@ public function flush($delay = 0)
                     $this->redis instanceof Predis\Client ? 0 : []
                 ));
             } else {
-                $result = $this->parse_redis_response($this->redis->flushdb());
+                if (defined('WP_REDIS_CLUSTER') && defined('WP_REDIS_CLIENT') && strcasecmp('pecl', WP_REDIS_CLIENT) === 0) {
+                    foreach($this->redis->_masters() as $master) {
+                        $result = $this->parse_redis_response($this->redis->flushdb($master));
+                    }
```

Comment:
I'd probably collect the `$results` and pass them to the `redis_object_cache_flush` action.